com.unity.render-pipelines.core/Runtime/PostProcessing/Shaders/LensFlareHDRPCommon.hlsl

```diff
@@ -0,0 +1,226 @@
+float4 _FlareData5; // x: Allow Offscreen, y: Edge Offset, z: Falloff
+
+#define _FlareColor             _FlareColorValue
+
+#define _LocalCos0              _FlareData0.x
+#define _LocalSin0              _FlareData0.y
+#define _PositionOffset         _FlareData0.zw
+
+#define _OcclusionRadius        _FlareData1.x
+#define _OcclusionSampleCount   _FlareData1.y
+#define _ScreenPosZ             _FlareData1.z
+#define _ScreenRatio            _FlareData1.w
+
+#define _ScreenPos              _FlareData2.xy
+#define _FlareSize              _FlareData2.zw
+
+#define _FlareRayOffset         _FlareData3.xy
+#define _FlareShapeInvSide      _FlareData3.z
+
+#define _FlareSDFRoundness      _FlareData4.x
+#define _FlareSDFPolyRadius     _FlareData4.y
+#define _FlareSDFPolyParam0     _FlareData4.z
+#define _FlareSDFPolyParam1     _FlareData4.w
+
+#define _OcclusionOffscreen     _FlareData5.x
+#define _FlareEdgeOffset        _FlareData5.y
+#define _FlareFalloff           _FlareData5.z
+
+float2 Rotate(float2 v, float cos0, float sin0)
+{
+    return float2(v.x * cos0 - v.y * sin0,
+                  v.x * sin0 + v.y * cos0);
+}
+
+#if FLARE_OCCLUSION
+float GetLinearDepthValue(float2 uv)
+{
+#ifdef HDRP_FLARE
+    float depth = LOAD_TEXTURE2D_X_LOD(_CameraDepthTexture, uint2(uv * _ScreenSize.xy), 0).r;
+
+    return 1.0 / (_ZBufferParams.z * depth + _ZBufferParams.w);
+#else
+    float depth = LOAD_TEXTURE2D_X(_CameraDepthTexture, uint2(uv * _ScreenSize.xy)).r;
+
+    return UNITY_Z_0_FAR_FROM_CLIPSPACE(depth);
+#endif
+}
+
+float GetOcclusion(float2 screenPos, float flareDepth, float ratio)
+{
+    if (_OcclusionSampleCount == 0.0f)
+        return 1.0f;
+
+    float contrib = 0.0f;
+    float sample_Contrib = 1.0f / _OcclusionSampleCount;
+    float2 ratioScale = float2(1.0f / ratio, 1.0);
+
+    for (uint i = 0; i < (uint)_OcclusionSampleCount; i++)
+    {
+        float2 dir = _OcclusionRadius * SampleDiskUniform(Hash(2 * i + 0 + 1), Hash(2 * i + 1 + 1));
+        float2 pos = screenPos + dir;
+        pos.xy = pos * 0.5f + 0.5f;
+#ifdef HDRP_FLARE
+        pos.y = 1.0f - pos.y;
+#else
+#endif
+        if (all(pos >= 0) && all(pos <= 1))
+        {
+            //float depth0 = LinearEyeDepth(SampleCameraDepth(pos), _ZBufferParams);
+            float depth0 = GetLinearDepthValue(pos);
+//#ifdef HDRP_FLARE
+            if (flareDepth < depth0)
+//#else
+//            if (flareDepth <= depth0)
+//#endif
+                contrib += sample_Contrib;
+        }
+        else if (_OcclusionOffscreen > 0.0f)
+        {
+            contrib += sample_Contrib;
+        }
+    }
+
+    return contrib;
+}
+#endif
+
+VaryingsLensFlare vert(AttributesLensFlare input, uint instanceID : SV_InstanceID)
+{
+    VaryingsLensFlare output;
+
+    float screenRatio = _ScreenRatio;
+
+    float4 posPreScale = float4(2.0f, 2.0f, 1.0f, 1.0f) * GetQuadVertexPosition(input.vertexID % 6) - float4(1.0f, 1.0f, 0.0f, 0.0);
+    output.texcoord = GetQuadTexCoord(input.vertexID % 6);
```

Comment:
I believe this won't work for gles2, but you can make it work similar to here: https://github.com/Unity-Technologies/Graphics/pull/4156/files#diff-a38bfa3320d1cc79f7e2f03bd3db8b2e82c77605cd0f44216dbb70e087c4892fR62-R80

Reply:
That doesn't to be a replacement. On the C# we should have both code path. One with Procedural and another one with a DrawMesh instead right?